gcloud-java-pubsub/src/main/java/com/google/cloud/pubsub/PubSub.java

```diff
@@ -0,0 +1,203 @@
+    }
+
+    public static ListOption pageToken(String pageToken) {
+      return new ListOption(Option.PAGE_TOKEN, pageToken);
+    }
+  }
+
+  final class PullOption implements Serializable {
+
+    private static final long serialVersionUID = -5220474819637439937L;
+
+    private final Option option;
+    private final Object value;
+
+    enum Option {
+      RETURN_IMMEDIATELY, MAX_MESSAGES
+    }
+
+    private PullOption(Option option, Object value) {
+      this.option = option;
+      this.value = value;
+    }
+
+    Option option() {
+      return option;
+    }
+
+    Object value() {
+      return value;
+    }
+
+    public static PullOption returnImmediatly() {
+      return new PullOption(Option.RETURN_IMMEDIATELY, true);
+    }
+
+    public static PullOption maxMessages(int maxMessages) {
+      return new PullOption(Option.MAX_MESSAGES, maxMessages);
+    }
+  }
+
+  /**
+   * A callback to process pulled messages.
+   * The message will be ack'ed upon successful return or nack'ed if exception is thrown.
+   */
+  interface PullCallback {
+    void process(Message message) throws Exception;
+  }
+
+  Topic create(TopicInfo topic);
+
+  Future<Topic> createAsync(TopicInfo topic);
+
+  // null if not found
+  Topic getTopic(String topic);
+
+  Future<Topic> getTopicAsync(String topic);
+
+  // false if not found
+  boolean deleteTopic(String topic);
+
+  Future<Boolean> deleteTopicAsync(String topic);
+
+  Page<Topic> listTopics(ListOption... options);
+
+  Future<AsyncPage<Topic>> listTopicsAsync(ListOption... options);
+
+  String publish(String topic, Message message);
+
+  Future<String> publishAsync(String topic, Message message);
+
+  List<String> publish(String topic, Message message, Message... messages);
+
+  Future<List<String>> publishAsync(String topic, Message message, Message... messages);
+
+  List<String> publish(String topic, Iterable<Message> messages);
+
+  Future<List<String>> publishAsync(String topic, Iterable<Message> messages);
+
+  Subscription create(SubscriptionInfo subscription);
+
+  Future<Subscription> createAsync(SubscriptionInfo subscription);
+
+  // null if not found
+  Subscription getSubscription(String subscription);
+
+  Future<Subscription> getSubscriptionAsync(String subscription);
+
+  void replacePushConfig(String subscription, PushConfig pushConfig);
+
+  Future<Void> replacePushConfigAsync(String subscription, PushConfig pushConfig);
+
+  // false if not found
+  boolean deleteSubscription(String subscription);
+
+  Future<Boolean> deleteSubscriptionAsync(String subscription);
+
+  Page<Subscription> listSubscriptions(ListOption... options);
+
+  Future<AsyncPage<Subscription>> listSubscriptionsAsync(ListOption... options);
+
+  Page<Subscription> listSubscriptions(String topic, ListOption... options);
+
+  Future<AsyncPage<Subscription>> listSubscriptionsAsync(String topic, ListOption... options);
+
+  List<ReceivedMessage> pull(String subscription, PullOption... options);
```

Comment:
This one does not suppose to auto-ack but rather auto renew ack deadline.

We could also make this method return an `Iterator<ReceivedMessage>` instead of a `List` and in that case
we will only renew the deadline for non consumed messages.  Considering that `ReceivedMessage` has also
a method to renew ack deadline I think that would be less confusing.